src/rebuild/scan.c

```diff
@@ -458,6 +472,76 @@ rebuild_container_scan_cb(daos_handle_t ih, vos_iter_entry_t *entry,
 	return rc;
 }
 
+/*
+ * TODO: This is a temporary feature to enable reintegration
+ * of targets with existing data. This will be removed when
+ * reintegration is able to make use of existing data
+ * (AKA Incremental Reintegration)
+ */
+static int
+reint_cont_destroy_cb(daos_handle_t ih, vos_iter_entry_t *entry,
+			  vos_iter_type_t type, vos_iter_param_t *iter_param,
+			  void *data, unsigned *acts) {
+	struct rebuild_scan_arg		*arg = data;
+	struct rebuild_tgt_pool_tracker *rpt = arg->rpt;
+	struct ds_cont_child		*ds_cont_child;
+	int				rc;
+
+	/* resync DTXs' status firstly. */
+	if (uuid_compare(arg->co_uuid, entry->ie_couuid) == 0) {
+		D_DEBUG(DB_REBUILD, DF_UUID" already scan\n",
+			DP_UUID(arg->co_uuid));
+		return 0;
+	}
+
+	rc = dtx_resync(iter_param->ip_hdl, rpt->rt_pool_uuid, entry->ie_couuid,
+			rpt->rt_rebuild_ver, true);
+	if (rc) {
+		D_ERROR(DF_UUID" dtx resync failed: rc %d\n",
+			DP_UUID(rpt->rt_pool_uuid), rc);
+		return rc;
+	}
+
+	rc = ds_cont_child_lookup(rpt->rt_pool_uuid, entry->ie_couuid,
+			&ds_cont_child);
+	if (rc != 0) {
+		D_ERROR(DF_UUID"Reintegration failed to lookup container child."
+				" rc: %d", DP_UUID(entry->ie_couuid), rc);
+		D_GOTO(out, rc);
+	}
+
+	rc = vos_cont_close(ds_cont_child->sc_hdl);
+	if (rc != 0) {
+		D_ERROR(DF_UUID"Reintegration failed to close container."
+				" rc: %d", DP_UUID(entry->ie_couuid), rc);
+		D_GOTO(out, rc);
+	}
+
+	rc = vos_cont_destroy(iter_param->ip_hdl, entry->ie_couuid);
+	if (rc != 0) {
+		D_ERROR(DF_UUID"Reintegration failed to destroy container."
+				" rc: %d", DP_UUID(entry->ie_couuid), rc);
+		D_GOTO(open, rc);
+	}
```

Comment:
I'm afraid that we can't call VOS API directly to close & destroy container, because the vos container is still being used by container module (like aggregation ULT), the safe way is to call a local container destroy API exported by container module (similar to ds_cont_local_open() which is being used by rebuild), the local container destroy API will be able to stop aggregation ULT and delete the container properly.